src/core/infer_request.cc

```diff
@@ -249,28 +249,54 @@ InferenceRequest::CopyAsNull(const InferenceRequest& from)
     new_input->SetData(data);
   }
 
-
   // Second pass
   size_t max_byte_size = 0;
   const std::string* max_input_name;
+
+  bool includes_type_bytes = false;
```

Comment:
Use a "zero_bytes_count" variable initialized to 0 and update it for TYPE_STRING to be the max str_byte_size. Then you only need to zero out the first 'zero_bytes_count' bytes of the buffer instead of zeroing the whole thing.